apps/nextjs/src/pages/index.tsx

```diff
@@ -135,27 +152,39 @@ const Home: NextPage = () => {
 export default Home;
 
 const AuthShowcase: React.FC = () => {
-  const { data: session } = api.auth.getSession.useQuery();
-
+  const supabaseClient = useSupabaseClient();
+  const user = useUser();
   const { data: secretMessage } = api.auth.getSecretMessage.useQuery(
     undefined, // no input
-    { enabled: !!session?.user },
+    { enabled: !!user },
   );
 
   return (
     <div className="flex flex-col items-center justify-center gap-4">
-      {session?.user && (
-        <p className="text-center text-2xl text-white">
-          {session && <span>Logged in as {session?.user?.name}</span>}
-          {secretMessage && <span> - {secretMessage}</span>}
-        </p>
+      {!user && (
+        <Auth
+          redirectTo="http://localhost:3000/"
+          appearance={{ theme: ThemeSupa }}
+          supabaseClient={supabaseClient}
+          providers={["apple", "github"]}
+          socialLayout="vertical"
+          onlyThirdPartyProviders
```

Comment:
there is some weird flash with the positioning here when only using the oauth providers. maybe just remove this component and use normal buttons with callbacks?

Reply:
Personally, I think we don't need to nail every single details at this point, but if you want to replace it, that would be nice too!